### PR TITLE
Add COLR table

### PR DESCRIPTION
**Pull request by dfrg**
Implements reading for the `COLR` table.

### CODE REVIEW

File: resources/codegen_inputs/colr.rs [hunk 1]
```diff
@@ -0,0 +1,859 @@
...
+    var_index_map_offset: BigEndian<Offset32>,
+    /// Offset to ItemVariationStore (may be NULL).
+    #[available(1)]
+    #[nullable]
+    item_variation_store_offset: BigEndian<Offset32>,
+}
+
+/// [BaseGlyphRecordArray](https://learn.microsoft.com/en-us/typography/opentype/spec/colr#baseglyph-and-layer-records) table
+#[read_args(num_base_glyph_records: u16)]
+table BaseGlyphRecordArray {
```
Comment (cmyr) [inline resources/codegen_inputs/colr.rs:45]: for these shim tables I might add a note in the docs explaining that it's a sort of a hack, and the rationale?

I also think in the medium term it probably makes sense to support 'offsets-to-non-tables' explicitly, but I'd have to think a bit about what this entails.

**Review by cmyr — APPROVED** [review]
This looks great, very encouraging!

File: resources/codegen_inputs/colr.rs (hunk 1 above)
Comment (cmyr) [reply resources/codegen_inputs/colr.rs:46]: The only place where it is dirty is in things like traversal, which is going to add an extra level of indirection? But I agree this case isn't too bad at all.

File: resources/codegen_inputs/colr.rs [hunk 2]
```diff
@@ -0,0 +1,861 @@
...
+    /// Set to 8.
+    #[format = 8]
+    format: BigEndian<u8>,
+    /// Offset to ColorLine table.
+    color_line_offset: BigEndian<Offset24<ColorLine>>,
+    /// Center x coordinate.
+    center_x: BigEndian<FWord>,
+    /// Center y coordinate.
+    center_y: BigEndian<FWord>,
+    /// Start of the angular range of the gradient, 180┬░ in
```
[the remark this answers is not in the document]
Comment (dfrg) [reply resources/codegen_inputs/colr.rs:409]: Looks like it. This was copy-pasted from the MS spec

[the remark this answers is not in the document]
Comment (cmyr) [reply resources/codegen_inputs/colr.rs:409]: fixed in #60

**Pull request merged**